tests/_runner.lua

```diff
@@ -136,18 +138,26 @@ function runner.run_steps(steps, options)
                 -- All steps finished, we are done.
                 runner.done()
             end
-        elseif result == false then
-            runner.done("Step "..step_as_string.." failed (returned false).")
         else
-            -- No result yet, run this step again.
-            wait = wait-1
-            if wait > 0 then
-                t.timeout = 0.1
-                t:again()
+            -- Append filename/lnum of failed step function.
+            local step_info = debug.getinfo(step_func)
+            local step_loc = string.format("%s:%d", step_info["short_src"], step_info["linedefined"])
+            step_as_string = step_as_string .. " ("..step_loc..")"
```

Comment:
I know that the `debug` library is supposedly slow, but I also doubt that this is relevant here, because this code is not executed that often, so:

Does it make sense to move this up to the code that sets up `step_as_string` so that all of it is together in one place? Or does that not make sense?

Reply:
I thought it would be only useful for failed tests, no?

Reply:
What I meant: Move this up to next to line 114 where `step_as_string` is defined the first time.

Reply:
Got that.  Then it is used for the "running …" msg(s) though also.  I.e. more noisy.

Reply:
Ah, okay.